verification/src/solidity/verifier.rs

```diff
@@ -311,10 +420,137 @@ impl Verifier {
     /// Verifies input data provided on initialization by comparing it
     /// with compiler output received when compiling source data locally.
     ///
-    /// If verification succeeds return [`Ok`], otherwise when verification
-    /// fails return an [`VerificationError`] inside [`Err`].
-    pub fn verify(&self, output: CompilerOutput) -> Result<(), VerificationError> {
-        todo!()
+    /// Iterates through all contracts received from local compilation and
+    /// returns [`VerificationSuccess`] with corresponding file path and contract name
+    /// if any contract  the contract that succeeds the verification. Otherwise, returns [`None`].
+    pub fn verify(&self, output: CompilerOutput) -> Option<VerificationSuccess> {
```

Comment:
is this used somewhere? or is it for future? function returns strange thing for me, but usage could explain it. If it's for the future, it's fine, I'll look into it when we'll use it

Reply:
I mean for me the part "if any contract the contract that succeeds the verification" is strange, not the return type

Reply:
> is this used somewhere? or is it for future? function returns strange thing for me, but usage could explain it. If it's for the future, it's fine, I'll look into it when we'll use it

The structure returns information obtained from the contract succeeded verification. It would be used further